lib/rubocop/cop/rspec/mixin/location.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Helper methods to location.
+      module Location
+        module_function
```

Comment:
How is it purposed to be used? As an includable module, by calling its methods, or both ways?

Does it make sense to name it `LocationHelp` to match `RangeHelp`/`commentsHelp`?

Reply:
I propose that you do not allow Include to do so. 
We believe that this will make it easier to follow the code, and we also use it at https://github.com/rubocop/rubocop-rspec/pull/1402#discussion_r995904977 .

So I changed it to define it as a private, singular method.

Also, it seems like a good to use `LocationHelp`. Thank you! I updated this PR.

Reply:
Oh, so you originally had `module_function` before @pirj convinced you to use `def self.foo`, after which I said what we should use `module_function` 😅

The downside of `module_function` and `extend(self)` is that the module’s methods can still be included into a class (as private vs public methods). At least I think that’s a downside.

I see that the Ruby source uses `module_function` a lot, and `extend self` quite little. It’s hard to tell how often they define a module only with class methods.

Reply:
>Oh, so you originally had module_function before @pirj convinced you to use def self.foo, after which I said what we should use module_function 😅

Yes, it is. But I thought if we as RuboCop have a flow that we basically use `module_function` we had better choose to do that. @pirj I would like to get your opinion.

Reply:
My rule of thumb is:
1. Use `module_function` for double-purpose modules that may be included or used as a standalone helper, and when included should not expose their methods
2. Use `extend self` for double-purpose modules that may be included or used as a standalone helper and would expose their methods when included
3. Use `class << self` for modules purposed to be used as helper that, or that are complex enough to have private methods not intended to be exposed in the module's API

In our case, I see no reason why we'd want `extend self` or `module_function` over `class << self`, but this is only a purist nitpick, and I don't insist on that at all.

Reply:
More on this https://github.com/rubocop/ruby-style-guide/issues/556

Reply:
> a purist nitpick

That could be the slogan of our club 😆 

I like your rules of thumb. It seems silly that RuboCop offers to enforce one style over the others when they clearly are not interchangeable.